impl/ocaml/sqlgg_traits.ml

```diff
@@ -113,16 +119,13 @@ module type M_io = sig
 
   module IO : Sqlgg_io.M
 
-  val finish_params : params -> result IO.future
-
-  val no_params : statement -> result IO.future
-
-  val select : [>`RO] connection -> string -> (statement -> result IO.future) -> (row -> unit) -> unit IO.future
-
-  val select_one_maybe : [>`RO] connection -> string -> (statement -> result IO.future) -> (row -> 'b) -> 'b option IO.future
-
-  val select_one : [>`RO] connection -> string -> (statement -> result IO.future) -> (row -> 'b) -> 'b IO.future
-
-  val execute : [>`WR] connection -> string -> (statement -> result IO.future) -> execute_response IO.future
+  include FNS
+    with type params := params
+    with type result := result
+    with type 'a io_future := 'a IO.future
+    with type 'a connection := 'a connection
+    with type statement := statement
+    with type row := row
+    with type execute_response := execute_response
```

Comment:
this slight visual sore prompts me to question why M_io is needed as separate module, instead of making IO-aware module a functor and then instantiate synchronous version with `type 'a future = 'a`, but not for this PR ofc